.github/workflows/ci-release.yml

```diff
@@ -28,23 +28,74 @@ on:
     uses: ./.github/workflows/lint.yml
 
   test:
     uses: ./.github/workflows/test.yml
 
-  # Make a release if this is a manually trigger job, i.e. workflow_dispatch
-  release:
+  # If this was a workflow dispatch event, we need to generate and push a tag
+  # for goreleaser to grab
+  version_bump:
     needs: [lint, test]
     runs-on: ubuntu-latest
-    if: ${{ github.event_name == 'workflow_dispatch' }}
     permissions: "write-all"
     steps:
       - uses: actions/checkout@v4
-      - name: Version Release
-        uses: celestiaorg/.github/.github/actions/version-release@v0.2.2
+      - name: Bump version and push tag
+        # Placing the if condition here is a workaround for needing to block
+        # on this step during workflow dispatch events but the step not
+        # needing to run on tags. If we had the if condition on the full
+        # version_bump section, it would skip and not run, which would result
+        # in goreleaser not running either.
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        uses: mathieudutour/github-tag-action@v6.0
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          default_bump: ${{ inputs.version }}
+
+  build_tags:
+    needs: version_bump
+    runs-on: ubuntu-latest
+    outputs:
+      BUILDTAGS: ${{ steps.build-tags.outputs.build-tags}}
+      BUILDTAGSCOMMASEP: ${{ steps.build-tags-comma-sep.outputs.build-tags-comma-sep}}
+    steps:
+      - uses: actions/checkout@v4
+      - id: build-tags
+        run: echo "build-tags=$(scripts/build-tags.sh | grep '^BUILD_TAGS=' | sed 's/^BUILD_TAGS=//')" >> "$GITHUB_OUTPUT"
+      - id: build-tags-comma-sep
+        run: echo "build-tags-comma-sep=$(scripts/build-tags.sh | grep '^BUILD_TAGS_COMMA_SEP=' | sed 's/^BUILD_TAGS_COMMA_SEP=//')" >> "$GITHUB_OUTPUT"
+
+  # Generate the release with goreleaser to include pre-built binaries
+  goreleaser:
+    needs: build_tags
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'workflow_dispatch' || (github.event_name == 'push' && contains(github.ref, 'refs/tags/')) }} # yamllint disable-line rule:line-length
+    permissions: "write-all"
+    steps:
+      - uses: actions/checkout@v3
+      - run: git fetch --force --tags
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.20
+      # Import the GPG key from github secrets to sign the binaries
+      # Uncomment if we want signing
+      # - name: Import GPG key
+      #   id: import_gpg
+      #   uses: crazy-max/ghaction-import-gpg@v4
+      #   with:
+      #     gpg_private_key: ${{ secrets.GPG_SIGNING_KEY }}
+      #     passphrase: ${{ secrets.GPG_PASSPHRASE }}
+      # Generate the binaries, release, and sign the checksum
+      - uses: goreleaser/goreleaser-action@v4
         with:
-          github-token: ${{secrets.GITHUB_TOKEN}}
-          version-bump: ${{inputs.version}}
+          distribution: goreleaser
+          version: latest
+          args: release --clean
```

Comment:
Just to clarify my understanding: this will attach the binaries to the release, right? But it won't mention them in the release notes, right?

Reply:
Correct, you can see the example of what it'll look like here https://goreleaser.com/quick-start/

also I tested this on a simple go project personally to verify the functionality.